google-authenticator-libpam 1.05 (new formula)

jabenninghoff opened the pull request:
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
macOS build of Google Authenticator example PAM module.

jabenninghoff commented:
The included test just checks "--help"; which isn't a good test, but since google-authenticator writes a file to the user's home directory in normal use, I couldn't see an easy way to write a better test. I'd appreciate pointers for that. Thanks!

ilovezfs commented:
>but since google-authenticator writes a file to the user's home directory in normal use

During `test do` the environment variable `HOME` is set equal to the `testpath` so it shouldn't be an issue unless it's ignoring `HOME`.

jabenninghoff commented:
Thanks, I'll work on an updated test. I also need to add a fix to the formula as the code looks for version 3 of libqrencode specifically (current version is 4).

ilovezfs commented:
Is version 4 support in master yet?

jabenninghoff commented:
I've tested with v4 and it appears to work with no issues. I'm going to submit a PR to look for `libqrencode.dylib` instead of `libqrencode.3.dylib`.

DomT4 commented on Formula/google-authenticator-libpam.rb:
```diff
@@ -0,0 +1,45 @@
+
+  def install
+    system "./bootstrap.sh"
+
```
Don't need this empty line.

DomT4 commented on Formula/google-authenticator-libpam.rb:
```diff
@@ -0,0 +1,45 @@
+
+  def caveats; <<-EOS.undent
+    Add 2-factor authentication for ssh:
+      echo "auth required /usr/local/lib/security/pam_google_authenticator.so" \\
```
You can't assume `/usr/local`. Use `#{opt_lib}`.

DomT4 commented on Formula/google-authenticator-libpam.rb:
```diff
@@ -0,0 +1,45 @@
+      echo "auth required /usr/local/lib/security/pam_google_authenticator.so" \\
+      "nullok" | sudo tee -a /etc/pam.d/sshd
+
+    (Or just manually edit /etc/pam.d/sshd)
```
Not sure any of the caveats below this are required to be honest, but I'll let ILZ weigh in.

jabenninghoff commented:
OK I've updated the PR:
* added fix to support current version (4) of libqrencode in homebrew (upstream PR submitted)
* removed extra whitespace
* changed caveat to use `#{opt_lib}`
* better test

Let me know if I should remove the caveat altogether. Thanks!

jabenninghoff commented:
OK,
I've removed the extra caveat as upon reflection, I agreed it was not necessary.
The upstream PR was accepted, and I've [requested](https://github.com/google/google-authenticator-libpam/issues/84) a new release. Upon release, I can update the version and remove the `inreplace`.

Is this OK to merge?

DomT4 commented on Formula/google-authenticator-libpam.rb:
```diff
@@ -0,0 +1,42 @@
+  depends_on "qrencode" => :recommended
+
+  def install
+    cd "src" do
```
Not a big thing but not really worth `cd`ing into a directory to execute an inreplace. Just do:
```ruby
inreplace "src/google-authenticator.c",
          "libqrencode.3.dylib",
          "libqrencode.4.dylib
```

ilovezfs commented:
We'll need to wait for that new release.

DomT4 commented:
Cool. You can ignore my advice then @jabenninghoff 😄.

jabenninghoff commented:
OK! I'll update the PR after the new release.

ilovezfs commented:
@BrewTestBot test this please

ilovezfs commented:
@stale why don't you learn how to read?

jabenninghoff commented:
@ilovezfs thanks for the bump on both this and the upstream request.

ilovezfs commented:
@jabenninghoff you're welcome! [1.05](https://github.com/google/google-authenticator-libpam/archive/1.05.tar.gz) is now out.

jabenninghoff commented:
Hi this is updated to 1.05. Should be ready to go, not sure why it hasn't built...

ilovezfs commented:
@BrewTestBot test this please

jabenninghoff commented:
Hi, @ilovezfs all tests have passed, let me know if there's anything else needed to merge!

ilovezfs commented:
Thanks for the :new: formula @jabenninghoff! :confetti_ball:
